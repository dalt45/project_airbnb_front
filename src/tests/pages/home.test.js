/*const multi = (a,b) => {
    return a*b;
}
test('should multiply 2*3 correctly', () = => {
    expect(multi(2,3)).toBe(6);
})

import React from 'react';
import {shallow} from 'enzyme';
import Home from '../../pages/home'

test('should render Home page correctly', () => {
    const wrapper = shallow(<Home />);
    // el nombre wrapper es por convenciÃ³n
    expect(wrapper).toMatchSnapshot();
})

*/
import React from 'react';
import { shallow } from'enzyme';
import Home from '../../pages/home';

test('should render Home page correctly', () => {
   const wrapper = shallow(<Home />);
   expect(wrapper.find('p').length).toBe(1);
   expect(wrapper).toMatchSnapshot();
})

test('should call onclick function', () => {
    const onClickSpy = jest.fn();
    const wrapper = shallow(<Home handleClick={onClickSpy}/>);
    wrapper.find('div').at(0).simulate('click');
    expect(onClickSpy.mock.calls.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
 })

