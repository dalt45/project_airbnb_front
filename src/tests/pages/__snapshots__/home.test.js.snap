// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call onclick function 1`] = `
<div
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  <p>
    Home
  </p>
</div>
`;

exports[`should render Home page correctly 1`] = `
<div>
  <p>
    Home
  </p>
</div>
`;
